File: backend/Models/Video.cs
namespace VideoTravelManager.Models;

public class Video
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;
    public int DurationSeconds { get; set; }
    public long FileSizeBytes { get; set; }
    public string? ThumbnailPath { get; set; }
    public DateTime ImportedAt { get; set; } = DateTime.UtcNow;
    public int PlayCount { get; set; }
    public DateTime? LastPlayedAt { get; set; }

    public ICollection<VideoCategory> VideoCategories { get; set; } = new List<VideoCategory>();
}

File: frontend/src/types/video.ts
export interface Video {
  id: number;
  title: string;
  thumbnailPath?: string;
  playCount: number;
  importedAt: string;
  categories: string[];
}

export interface VideoDetail extends Video {
  filePath: string;
  durationSeconds: number;
  fileSizeBytes: number;
  lastPlayedAt?: string;
}
